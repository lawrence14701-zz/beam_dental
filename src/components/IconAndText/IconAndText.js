import React from "react"
import FiberManualRecordIcon from "@material-ui/icons/FiberManualRecord"
import useStyles from "./style"
import cx from "classnames"

const IconAndText = (props) => {
  const classes = useStyles()
  const { text, color, num } = props
  return (
    <div>
      <FiberManualRecordIcon
        className={cx(
          color && {
            [classes.blue]: color === "blue",
            [classes.green]: color === "green",
            [classes.pink]: color === "pink",
          }
        )}
      />
      <span>{num}</span>
      <span>{text}</span>
    </div>
  )
}

export default IconAndText
