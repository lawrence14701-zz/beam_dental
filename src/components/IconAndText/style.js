import { makeStyles } from "@material-ui/core/styles"

const useStyles = makeStyles((theme) => ({
  blue: {
    color: theme.palette.common.blue,
  },
  pink: {
    color: theme.palette.common.pink,
  },
  green: {
    color: theme.palette.common.green,
  },
}))

export default useStyles
