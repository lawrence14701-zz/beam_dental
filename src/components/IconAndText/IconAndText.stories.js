import React from 'react';
import IconAndText from './IconAndText';
import { ThemeProvider } from '@material-ui/styles';
import muiTheme from '../../Theme/theme'


export default {
	title: 'IconAndText',
	component: IconAndText,
	args: { text: 'brushes', 'num': '2' },
	argTypes: { onClick: { action: 'clicked' } },
	decorators: [(Story) => <ThemeProvider theme={muiTheme}><Story/></ThemeProvider>]

};

const Template = (args) => <IconAndText {...args} />;

export const Blue = Template.bind({});
export const Pink = Template.bind({});
export const Green = Template.bind({});

Blue.args = {
	color: 'blue',
};

Pink.args = {
	color: 'pink',
};

Green.args = {
	color: 'greens',
};
