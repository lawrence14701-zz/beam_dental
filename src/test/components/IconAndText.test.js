import IconAndText from "../../components/IconAndText/IconAndText"
import { shallow } from "enzyme"
import React from "react"
import "../jestsetup"

function createTestProps() {
  return {
    text: "testing",
    color: "blue",
    num: "2",
  }
}

test("render a circle icon with text", () => {
  const props = createTestProps()
  const wrapper = shallow(<IconAndText {...props} />)
  expect(wrapper).toMatchSnapshot()
})
