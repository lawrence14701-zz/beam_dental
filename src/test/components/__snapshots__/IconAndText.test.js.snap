// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render a circle icon with text 1`] = `ShallowWrapper {}`;
